### PR TITLE
fix: Allow application document generation when child DoB is in future (FPLA-2896)

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
https://tools.hmcts.net/jira/browse/FPLA-2896

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java [hunk 1]
```diff
@@ -13,27 +12,35 @@ private AgeDisplayFormatHelper() {
...
-        int years = period.getYears();
-        int months = period.getMonths();
-        int days = period.getDays();
+        final Period period = Period.between(dateOfBirth, LocalDate.now());
+        final int years = period.getYears();
+        final int months = period.getMonths();
+        final int days = period.getDays();
 
+        if (period.isNegative()) {
+            return "0 years old";
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java:22]: Does the last return not cover this?

**Review by lewisbirks — CHANGES_REQUESTED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java (hunk 1 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java:22]: last return says '0 days old' not '0 years old' and is used when child born today

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java:22]: Could it not be updated to just return the same? Or do we want the explicit difference?

Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java:22]: this is what was agreed with Sara today on standup.

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/AgeDisplayFormatHelper.java:22]: sure thing

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**